add BackendConfigPolicy to TLS doc (#414)
kgateway-dev/kgateway.dev: content/docs/main/security/backend-tls.md

   nginx   1/1     Running   0          9s
   ```
   
### Create a TLS policy {#create-backend-tls-policy}

> Create a TLS policy for the NGINX workload. You can use the Gateway API BackendTLSPolicy for simple, one-way TLS connections. For more advanced TLS connections or simply to reduce the number of resources if you use other backend connections, create a BackendConfigPolicy instead.

Author: So the BackendConfigPolicy is what does not work with agentgateway, not the BackendTLSPolicy...although in this particular instance, neither work with agentgateway right now 🙃

{{< tabs tabTotal="2" items="BackendConfigPolicy,BackendTLSPolicy" >}}
{{% tab tabName="BackendConfigPolicy" %}}

> {{< reuse "docs/snippets/proxy-kgateway.md" >}}

Reviewer: actually we might want to say here that this works for envoy and agw

> {{< reuse "docs/snippets/proxy-kgateway.md" >}}

Author: same comment as above

   EOF
   ```

   {{< reuse "docs/snippets/review-table.md" >}} For more information, see the [{{< reuse "docs/snippets/k8s-gateway-api-name.md" >}} docs](https://gateway-api.sigs.k8s.io/api-types/backendtlspolicy/).

   | Setting | Description |
   |---------|-------------|

>    | `targetRefs` | The service that you want the Gateway to originate a TLS connection to, such as the NGINX server. <br><br>**Agentgateway proxies**: Even if you use a Backend for selector-based destinations, you still need to target the backing Service and the `sectionName` of the port that you want the policy to apply to.  |

Reviewer: ```suggestion
   | `targetRefs` | The service that you want the Gateway to originate a TLS connection to, such as the NGINX server. <br><br>**Agentgateway proxies**: This policy cannot be applied to an agentgateway proxy. |
```

   EOF
   ```
   

> 2. Create a TLS policy that originates a TLS connection to the Backend that you created in the previous step. To originate the TLS connection, you use known trusted CA certificates. You can use the Gateway API BackendTLSPolicy for simple, one-way TLS connections. For more advanced TLS connections or simply to reduce the number of resources if you use other backend connections, create a BackendConfigPolicy instead.

Reviewer: there are more instances in the external service section that need to be rephrased, such as the short description. and then depending on what you do for the wording further up, you can apply here too
